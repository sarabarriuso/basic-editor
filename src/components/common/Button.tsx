import classnames from 'classnames';
import React from 'react';

export interface IButtonProps {
  onClick: (e: React.MouseEvent<HTMLButtonElement>) => void;
  text: string;
  className?: string;
  disabled?: boolean;
  variant?: ButtonVariant;
}
export enum ButtonVariant {
  Primary = 'Primary',
  Secondary = 'Secondary',
  Tertiary = 'Tertiary',
}

const Button: React.FC<IButtonProps> = ({
  className,
  onClick,
  text,
  disabled,
  variant = ButtonVariant.Primary,
}) => {
  const DEFAULT_CLASSNAME = 'button';

  const buttonClassName: string = classnames(DEFAULT_CLASSNAME, className, {
    [`${DEFAULT_CLASSNAME}--secondary`]: variant === ButtonVariant.Secondary,
    [`${DEFAULT_CLASSNAME}--tertiary`]: variant === ButtonVariant.Tertiary,
  });

  return (
    <button
      className={buttonClassName}
      type='submit'
      onClick={onClick}
      disabled={disabled}
    >
      {text}
    </button>
  );
};

export default Button;
