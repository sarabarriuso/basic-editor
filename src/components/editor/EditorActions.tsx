import React from 'react';

import t from '../../language';
import Button, { ButtonVariant } from '../common/Button';

interface IEditorActionsProps {
  createShapeCallback: () => void;
  removeShapeCallback: () => void;
  shapeToDelete: boolean;
}

const EditorActions: React.FC<IEditorActionsProps> = ({
  createShapeCallback,
  removeShapeCallback,
  shapeToDelete,
}) => {
  const DEFAULT_CLASSNAME = 'editor-actions';

  return (
    <div className={`${DEFAULT_CLASSNAME}`}>
      <Button
        text={t.text('app.editor.actions.add')}
        onClick={() => {
          createShapeCallback();
        }}
      />
      <Button
        text={t.text('app.editor.actions.remove')}
        onClick={() => {
          removeShapeCallback();
        }}
        disabled={!shapeToDelete}
        variant={ButtonVariant.Secondary}
      />
    </div>
  );
};

export default EditorActions;
