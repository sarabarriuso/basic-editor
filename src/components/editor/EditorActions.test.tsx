import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Button from '../common/Button';
import EditorActions from './EditorActions';

export default describe('EditorActions Tests', () => {
  it('Editor actions renders successfully', () => {
    // Arrange
    // Act
    const tree = shallow(
      <EditorActions
        createShapeCallback={() => {
          return;
        }}
        removeShapeCallback={() => {
          return;
        }}
        shapeToDelete={false}
      />,
    );

    //Assert
    expect(toJson(tree)).toMatchSnapshot();
  });

  it('Clicking Add node button triggers createShapeCallback', () => {
    // Arrange
    let createShapedCalled = false;
    // Act
    const tree = shallow(
      <EditorActions
        createShapeCallback={() => (createShapedCalled = true)}
        removeShapeCallback={() => {
          return;
        }}
        shapeToDelete={false}
      />,
    );

    const addNodeButton = tree.find(Button).first();
    expect(addNodeButton.props().text).toBe('Add node');
    addNodeButton.simulate('click');

    //Assert
    expect(createShapedCalled).toBeTruthy();
  });

  it('Clicking Remove node button triggers removeShapeCallback', () => {
    // Arrange
    let removeShapedCalled = false;
    // Act
    const tree = shallow(
      <EditorActions
        createShapeCallback={() => {
          return;
        }}
        removeShapeCallback={() => (removeShapedCalled = true)}
        shapeToDelete={false}
      />,
    );

    const addNodeButton = tree.find(Button).last();
    expect(addNodeButton.props().text).toBe('Remove node');
    addNodeButton.simulate('click');

    //Assert
    expect(removeShapedCalled).toBeTruthy();
  });
});
