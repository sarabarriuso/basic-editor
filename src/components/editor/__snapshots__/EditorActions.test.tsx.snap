// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditorActions Tests Editor actions renders successfully 1`] = `
<div
  className="editor-actions"
>
  <Button
    onClick={[Function]}
    text="Add node"
  />
  <Button
    disabled={true}
    onClick={[Function]}
    text="Remove node"
    variant="Secondary"
  />
</div>
`;
